### PR TITLE
Globally set `Content-Type` header in integration tests

### PR DESCRIPTION
**Pull request by brianstrauch**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
A small improvement to make writing integration tests slightly easier. Allows us to delete all instances of `w.Header().Set("Content-Type", "application/json")`.

Test & Review
-------------
Integration tests still pass

### CODE REVIEW

File: test/test-server/ccloudv1_handlers.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 )
 
 // Handler for "/api/organizations/"
-func (c *CloudRouter) HandlePriceTable(t *testing.T) http.HandlerFunc {
+func handlePriceTable(t *testing.T) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
```
Comment (MuweiHe) [inline test/test-server/ccloudv1_handlers.go:18]: some handlers don't have the router receiver? How is `router.HandleFunc(route.path, route.handler(t))` working with them?

File: test/test-server/middleware.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package testserver
+
+import "net/http"
+
+func defaultHeaderMiddleware(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
```
Comment (MuweiHe) [inline test/test-server/middleware.go:7]: 👍

File: test/test-server/ccloudv1_handlers.go (hunk 1 above)
Comment (brianstrauch) [reply test/test-server/ccloudv1_handlers.go:18]: I'm iterating through a list of `route`s, which have `path` and `handler` fields. (Note the difference between `route` and `router`)

File: test/test-server/ccloud_router.go [hunk 3]
```diff
@@ -9,63 +9,44 @@ import (
...
-}
+	for _, route := range ccloudHandlers {
+		router.HandleFunc(route.path, route.handler(t))
+	}
 
-func (c CloudRouter) addV2AlphaRoutes(t *testing.T) {
-	c.HandleFunc(v2alphaAuthenticate, c.HandleV2Authenticate(t))
-	c.addRoutesAndReplies(t, v2Base, v2RoutesAndReplies)
+	router.HandleFunc("/api/me", handleMe(t, isAuditLogEnabled))
+	router.HandleFunc("/api/schema_registries", router.HandleSchemaRegistries(t))
```
Comment (sgagniere) [inline test/test-server/ccloud_router.go:46]: This one being the odd one out is unfortunate.

**Review by sgagniere — APPROVED** [review]
+1

File: test/test-server/ccloud_router.go (hunk 3 above)
Comment (brianstrauch) [reply test/test-server/ccloud_router.go:46]: Yeah, we can rewrite this later!

**Pull request merged**